parquet-common/src/main/java/org/apache/parquet/bytes/CapacityByteArrayOutputStream.java

```diff
@@ -164,6 +164,12 @@ public CapacityByteArrayOutputStream(int initialSlabSize, int maxCapacityHint, B
   private void addSlab(int minimumSize) {
     int nextSlabSize;
 
+    if (bytesUsed + minimumSize < 0) {
```

Comment:
nit: replace it with `Math.addExact` to hand over overflow check. Then we get the chance to check the 2GB hard limit before it becoming negative

Reply:
Updated